dps310/dps310.go

```diff
@@ -0,0 +1,539 @@
+package dps310
+
+// Datasheet:
+// https://www.infineon.com/dgdl/Infineon-DPS310-DataSheet-v01_02-EN.pdf?fileId=5546d462576f34750157750826c42242
+
+import (
+	"fmt"
+	"machine"
+	"math"
+	"time"
+)
+
+// hardware interfacing with a DPS310
+type Config struct {
+	SPI        *machine.SPI
+	CSD        machine.Pin
+	SPI_Config machine.SPIConfig
+}
+
+type DPS310 struct {
+	c0, c1, c01, c00, c10, c11, c20, c21, c30 float32
+	tempScale, pressureScale                  float32
+	tempRate, pressureRate                    Rate
+	tempOs, pressureOs                        Precision
+	seaLevelPressure                          float32
+	mode                                      Mode
+	rwBuf                                     [4]byte
+	spi                                       *machine.SPI
+	cs                                        machine.Pin
+}
+
+func NewSPI(c Config) (*DPS310, error) {
+	d := &DPS310{
+		spi: c.SPI,
+		cs:  c.CSD,
+	}
+
+	err := d.spi.Configure(c.SPI_Config)
```

Comment:
This should probably validate, that the SPI mode is `Mode3` and the frequency is equal to or below the maximum supported 10 MHz